evap/staff/importers.py

```diff
@@ -97,28 +105,111 @@ def store_in_database(self, vote_start_datetime, vote_end_date, semester):
+                    self.name_en if name_collision == NameCollision.NAME_EN else self.name_de,
+                    "EN" if name_collision == NameCollision.NAME_EN else "DE",
+                )
+            )
+            return False
+
+        course = Course.objects.filter(
             semester=semester,
+            name_en=self.name_en,
+            name_de=self.name_de,
+        ).first()
+
+        if course is None:
+            # This can only happen if the name collisions happened in different courses
+            importer.errors[ImporterError.COURSE].append(
+                _("Course {} (EN) and Course {} (DE) do already exist separately in this semester.").format(
+                    self.name_en, self.name_de
+                )
+            )
+            return False
+
+        mismatches, has_wrong_evaluation_count = self.check_existing_course_attributes(course)
+
+        if mismatches or has_wrong_evaluation_count:
+            messages = (
+                [
+                    "the {} {}".format(
+                        " and ".join(mismatches),
+                        ngettext("does not match", "do not match", len(mismatches)),
+                    )
+                ]
+                if mismatches
+                else []
+            )
+            messages += ["the course must have exactly one evaluation"] if has_wrong_evaluation_count else []
+            importer.errors[ImporterError.COURSE].append(
+                _("Course {} ({}) does already exist in this semester but {}.").format(
+                    self.name_en,
+                    self.name_de,
+                    " and ".join(messages),
+                )
+            )
+        else:
+            self.existing_course = course
+
+        return self.existing_course is not None
+
+    def check_name_collision(self, semester):
+        flag = NameCollision.NONE
+        flag |= NameCollision(
+            Course.objects.filter(semester=semester, name_en=self.name_en).exists() and NameCollision.NAME_EN
         )
-        course.save()
-        course.responsibles.set([responsible_dbobj])
-        course.degrees.set(self.degrees)
-        evaluation = Evaluation(
-            vote_start_datetime=vote_start_datetime,
-            vote_end_date=vote_end_date,
-            course=course,
-            wait_for_grade_upload_before_publishing=self.is_graded,
-        )
-        evaluation.save()
-        evaluation.contributions.create(
-            evaluation=evaluation,
-            contributor=responsible_dbobj,
-            role=Contribution.Role.EDITOR,
-            textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
+        flag |= NameCollision(
+            Course.objects.filter(semester=semester, name_de=self.name_de).exists() and NameCollision.NAME_DE
         )
+        return flag
+
+    def check_existing_course_attributes(self, course):
+        mismatches = []
+        has_wrong_evaluation_count = False
```

Comment:
The only branch that sets this to `True` returns immediately, so it can just be replaced by `True` and `False` in the two return statements directly